base/src/error.rs

```diff
@@ -67,6 +68,13 @@ pub enum BlockError {
     /// Io error.
     #[fail(display = "{}", _0)]
     Io(std::io::Error),
+    // RwLock error
+    #[fail(display = "{}", _0)]
+    RwLockReadGuard(failure::Error),
+    #[fail(display = "{}", _0)]
+    PoisonError(failure::Error),
+    #[fail(display = "{}", _0)]
+    RwLockWriteGuard(failure::Error),
```

Comment:
I do not understand how previous implementation avoided this since `BufStore` uses `RWLock`, I suppose `blocks.read().await` somehow avoids having to deal with `RwLockReadGuard` but I don't understand how.
I would love if someone could explain that to me.

I am also not sure if there is a better way to go about it.

Reply:
the store was always async, it's merely the codec I meant in #19

Reply:
Got it. @dvc94ch should I add a change to make store async again ? (Asking because I noticed your other pull requests are based off this).

Reply:
I removed the part that makes the store sync from the PR. I think my PR superceedes this one and we can close it, if you're ok with it.

Reply:
Ah ok, I must have missed those changes. 👍